DAC/src/ringbuffer.c

```diff
@@ -0,0 +1,65 @@
+/*
+ * ringbuffer.c
+ *
+ *  Created on: Feb 6, 2018
+ *      Author: sharana
+ *      Description: Program to initiate a ring buffer.
+ */
+#include"ringbuffer.h"
+#include<inttypes.h>
+/*
+ * Here we have a method put characters into the buffer.
+ * @param buffer it is a structure type to access the correct ring buffer.
+ * @param element it is the character.
+ */
+void put(RingBuffer* buffer, char element){
+	//Check space
+	 if(hasSpace(&buffer)== 1)
+	    {	//adding element to the buffer
+	        buffer->buffer[buffer->put] = element;
+	        buffer->put = (buffer->put + 1) % BUF_SIZE;
+	    }else{
+	    	//it it is at the get
+	            buffer->get = (buffer->get + 1) % BUF_SIZE;
+	            buffer->buffer[buffer->get] = element;
+
+	        }
+	    }
+/*
+ * Here we have a method to get and remove characters into the buffer.
+ * @param buffer it is a structure type to access the correct ring buffer.
+ */
+char get(RingBuffer* buffer){
+	//check space
+	if(hasElement(&buffer)== 1){
+		//get character from the  buffer
+	char element = buffer->buffer[buffer->get];
+	//adjusting get
+	buffer->get = (buffer->get+1) % BUF_SIZE;
+	return element;
+
+	}
```

Comment:
What should the get function return if there really isn't an element? All paths should have a return in C, because C will only give a compiler warning, if anything.